engine/execution/migration/badgerpebble.go

```diff
@@ -0,0 +1,202 @@
+func MigrateLastSealedExecutedResultToPebble(logger zerolog.Logger, badgerDB *badger.DB, pebbleDB *pebble.DB, state protocol.State, rootSeal *flow.Seal) error {
+	// only run the migration for mainnet26 and testnet52
+	sporkID := state.Params().SporkID()
+	chainID := state.Params().ChainID()
+	if chainID == flow.Mainnet || chainID == flow.Testnet {
+		if sporkID != mainnet26SporkID && sporkID != testnet52SporkID {
+			logger.Warn().Msgf("spork ID %v is not Mainnet26SporkID %v or Testnet52SporkID %v, skip migration",
+				sporkID, mainnet26SporkID, testnet52SporkID)
+			return nil
+		}
+	}
+
+	bdb := badgerimpl.ToDB(badgerDB)
+	pdb := pebbleimpl.ToDB(pebbleDB)
+	lg := logger.With().Str("module", "badger-pebble-migration").Logger()
+
+	// bootstrap pebble database
+	bootstrapper := bootstrap.NewBootstrapper(logger)
+	commit, bootstrapped, err := bootstrapper.IsBootstrapped(pdb)
+	if err != nil {
+		return fmt.Errorf("could not query database to know whether database has been bootstrapped: %w", err)
+	}
+
+	if !bootstrapped {
+		err = bootstrapper.BootstrapExecutionDatabase(pdb, rootSeal)
+		if err != nil {
+			return fmt.Errorf("could not bootstrap pebble execution database: %w", err)
+		}
+	}
+
+	// get last sealed and executed block in badger
+	lastExecutedSealedHeightInBadger, err := latest.LatestSealedAndExecutedHeight(state, bdb)
+	if err != nil {
+		return fmt.Errorf("failed to get last executed sealed block: %w", err)
+	}
+
+	// read all the data and save to pebble
+	header, err := state.AtHeight(lastExecutedSealedHeightInBadger).Head()
+	if err != nil {
+		return fmt.Errorf("failed to get block at height %d: %w", lastExecutedSealedHeightInBadger, err)
+	}
+
+	blockID := header.ID()
+
+	lg.Info().Msgf(
+		"migrating last executed and sealed block %v (%v) from badger to pebble",
+		header.Height, blockID)
+
+	// create badger storage modules
+	badgerResults, badgerCommits := createStores(bdb)
+	// read data from badger
+	result, commit, err := readResultsForBlock(blockID, badgerResults, badgerCommits)
+
+	if err != nil {
+		return fmt.Errorf("failed to read data from badger: %w", err)
+	}
+
+	// create pebble storage modules
+	pebbleResults, pebbleCommits := createStores(pdb)
+
+	// store data to pebble in a batch update
+	err = pdb.WithReaderBatchWriter(func(batch storage.ReaderBatchWriter) error {
+		var existingExecuted flow.Identifier
+		err = operation.RetrieveExecutedBlock(batch.GlobalReader(), &existingExecuted)
+		if err == nil {
+			// there is an executed block in pebble, compare if it's newer than the badger one,
+			// if newer, it means EN is storing new results in pebble, in this case, we don't
+			// want to update the executed block with the badger one.
+
+			header, err := state.AtBlockID(existingExecuted).Head()
+			if err != nil {
+				return fmt.Errorf("failed to get block at height %d from badger: %w", lastExecutedSealedHeightInBadger, err)
+			}
+
+			if header.Height > lastExecutedSealedHeightInBadger {
```

Comment:
To simplify code, maybe move this block of code (retrieving executed block height from pebble and comparing it with executed block height from badger) to outside of `pdb.WithReaderBatchWriter()`.  

We can return early and avoid reading more block data from badger if migration isn't needed (`header.Height > lastExecutedSealedHeightInBadger`).